compiler/cpp/test/parser/ParserTest.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 Compose Generator Contributors. All rights reserved.
+
+#include <parser/Parser.h>
+#include <gtest/gtest.h>
+#include "../util/LanguageDescriptor.h"
+#include "../../lib/json/json.hpp"
+#include "../util/FileReader.h"
+
+using json = nlohmann::json;
+
+struct ParserParams {
+    const std::string fileName;
+};
+
+const ParserParams PARSER_TEST_PARAMETERS[] = {
+        {
+            "has-block"
+        },
+        {
+            "comp-block"
+        },
+        {
+            "contains-block"
+        }
+};
+
+class ParserTests : public ::testing::TestWithParam<ParserParams> {};
+
+TEST_P(ParserTests, TestParserWithExpectToken) {
```

Comment:
Nice, these parameterized tests are great 👌 I get why we need the `serialize()` methods now ;)